src/components/ec-inline-input-field/ec-inline-input-field.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <div
+    class="tw-truncate"
+    data-test="ec-inline-input-field"
+  >
+    <div class="tw-mini-header">{{ label }}</div>
+    <div v-if="isEditable">
+      <ec-inline-input-field-read-only
+        v-if="isReadOnly"
+        :value="value"
+        :gain-focus="gainFocus"
+        @edit="edit"
+      />
+      <ec-inline-input-field-edit
+        v-else-if="isEditing"
+        :original-value="value"
+        :status="status"
+        @cancel="cancel"
+        @submit="submit"
+      />
+      <ec-inline-input-field-loading
+        v-else
+        :value="innerValue"
+      />
+    </div>
+    <slot v-else />
+  </div>
+</template>
+
+<script>
+import { EDITING, LOADING, READ_ONLY } from '@/enums/input-status';
+import EcInlineInputFieldEdit from './components/edit';
+import EcInlineInputFieldLoading from './components/loading';
+import EcInlineInputFieldReadOnly from './components/read-only';
+
+export default {
+  name: 'EcInlineInputField',
+  components: {
+    EcInlineInputFieldEdit,
+    EcInlineInputFieldLoading,
+    EcInlineInputFieldReadOnly,
+  },
+  props: {
+    label: {
+      default: '',
+      type: String,
+    },
+    isEditable: {
+      type: Boolean,
+      default: false,
+    },
+    status: {
```

Comment:
All components we have are using `is-loading` prop. I'd continue with that. My problem is how to use this code in my component:
```
<ec-inline-input-field :status="LOADING">
...
import { LOADING, EDITING } from '@/enums';
...
export default {
  data() {
    return { LOADING, EDITING }
  }
}
```
I'd prefer two booleans I can set in the template:
```
<ec-inline-input-field :is-loading="true" :is-editing="true">
```